komodoenv/update.py

```diff
@@ -335,46 +305,68 @@ def rewrite_executable(path, python, text):
         and newline_pos >= 0
         and text[:newline_pos].find(b"python") >= 0
     ):
-        return ensure_binary(
-            "#!{python}{rest}".format(
-                python=python, rest=ensure_str(text[newline_pos:])
-            )
-        )
+        return b"#!" + python.encode("utf8") + text[newline_pos:]
 
-    return ensure_binary(
+    return (
         dedent(
             """\
     #!/bin/bash
     export LD_LIBRARY_PATH={libs}${{LD_LIBRARY_PATH:+:${{LD_LIBRARY_PATH}}}}
     exec -a "$0" "{prog}" "$@"
     """
         ).format(libs=libs, prog=path)
-    )
+    ).encode("utf8")
 
 
-def update_bins(srcpath, dstpath):
+def update_bins(srcpath: str, dstpath: str):
     python = os.path.join(dstpath, "root", "bin", "python")
     shimdir = os.path.join(dstpath, "root", "shims")
     if os.path.isdir(shimdir):
         shutil.rmtree(shimdir)
 
     os.mkdir(shimdir)
     for name in os.listdir(os.path.join(srcpath, "root", "bin")):
+        if os.path.isfile(os.path.join(dstpath, "root", "bin")):
```

Comment:
Not sure if I understand really how the updating works, but why this if is inside the for loop?

Reply:
That is a bug most likely, thanks.